infra/prover/index.ts

```diff
@@ -0,0 +1,338 @@
+            'ecr:BatchGetImage',
+            'logs:CreateLogStream',
+            'logs:PutLogEvents',
+          ],
+          Resource: '*',
+        },
+        {
+          Effect: 'Allow',
+          Action: ['secretsmanager:GetSecretValue', 'ssm:GetParameters'],
+          Resource: [privateKeySecret.arn, bonsaiSecret.arn],
+        },
+      ],
+    },
+  });
+
+  const brokerEcr = new awsx.ecr.Repository(serviceName, {
+    lifecyclePolicy: {
+      rules: [
+        {
+          description: 'Delete untagged images after N days',
+          tagStatus: 'untagged',
+          maximumAgeLimit: 7,
+        },
+      ],
+    },
+    forceDelete: true,
+    name: serviceName,
+  });
+
+  const authToken = aws.ecr.getAuthorizationTokenOutput({
+    registryId: brokerEcr.repository.registryId,
+  });
+
+  // Optionally add in the sccache s3 creds to the build ctx
+  let buildSecrets = {};
+  if (ciCacheSecret !== undefined) {
+    const cacheFileData = ciCacheSecret.apply((filePath: any) => fs.readFileSync(filePath, 'utf8'));
+    buildSecrets = {
+      ci_cache_creds: cacheFileData,
+    };
+  }
+  if (githubTokenSecret !== undefined) {
+    buildSecrets = {
+      ...buildSecrets,
+      githubTokenSecret
+    }
+  }
+
+  const dockerTagPath = pulumi.interpolate`${brokerEcr.repository.repositoryUrl}:${dockerTag}`;
+
+  // TODO use this to build? Can this be used in ec2?
+  // const image = new docker_build.Image(serviceName, {
+  //   tags: [dockerTagPath],
+  //   context: {
+  //     location: dockerDir,
+  //   },
+  //   platforms: ['linux/amd64'],
+  //   push: true,
+  //   dockerfile: {
+  //     location: `${dockerDir}/dockerfiles/broker.dockerfile`,
+  //   },
+  //   buildArgs: {
+  //     S3_CACHE_PREFIX: 'private/boundless/rust-cache-docker-Linux-X64/sccache',
+  //   },
+  //   secrets: buildSecrets,
+  //   cacheFrom: [
+  //     {
+  //       registry: {
+  //         ref: pulumi.interpolate`${brokerEcr.repository.repositoryUrl}:cache`,
+  //       },
+  //     },
+  //   ],
+  //   cacheTo: [
+  //     {
+  //       registry: {
+  //         imageManifest: true,
+  //         ociMediaTypes: true,
+  //         ref: pulumi.interpolate`${brokerEcr.repository.repositoryUrl}:cache`,
+  //       },
+  //     },
+  //   ],
+  //   registries: [
+  //     {
+  //       address: brokerEcr.repository.repositoryUrl,
+  //       password: authToken.password,
+  //       username: authToken.userName,
+  //     },
+  //   ],
+  // });
+
+  const brokerSecGroup = new aws.ec2.SecurityGroup(serviceName, {
+    name: serviceName,
+    vpcId: vpcId,
+    egress: [
+      {
+        fromPort: 0,
+        toPort: 0,
+        protocol: '-1',
+        cidrBlocks: ['0.0.0.0/0'],
+        ipv6CidrBlocks: ['::/0'],
+      },
+    ],
+  });
+
+  // Set up CloudWatch logging for the EC2 instance
+  const brokerLogGroup = new aws.cloudwatch.LogGroup(serviceName, {
+    name: serviceName,
+    retentionInDays: 0,
```

Comment:
If you select 0, the events in the log group are always retained and never expire. Is this what we want?